### PR TITLE
Add aws_polling config option to override env variables

### PR DESCRIPTION
**Pull request by sylviamoss**
This adds a new option to the amazon builders for setting the AWS Waiter configuration. This is already possible via environment variables, but adding the options will allow overriding the env value for specific builds.

Closes https://github.com/hashicorp/packer/issues/9756

### CODE REVIEW

File: builder/amazon/common/state.go [hunk 1]
```diff
@@ -38,12 +40,45 @@ type StateChangeConf struct {
...
+// aws_polling {
+//	 delay_seconds = 30
+//	 max_attempts = 50
+// }
+// ```
+//
+type AWSPollingConfig struct {
+	// Specifies the maximum number of attempts the waiter will check for resource state.
+	// This value can also be set via the AWS_MAX_ATTEMPTS.
+	// If both option and environment variable are set, the max_attepmts will be considered over the AWS_MAX_ATTEMPTS.
```
Comment (azr) [inline builder/amazon/common/state.go:66]:
```suggestion
	// If both option and environment variable are set, the max_attempts will be considered over the AWS_MAX_ATTEMPTS.
```

File: builder/amazon/common/state.go [hunk 2]
```diff
@@ -38,12 +40,45 @@ type StateChangeConf struct {
...
+	// If none is set, defaults to AWS waiter default which is 40.
```
Comment (azr) [inline builder/amazon/common/state.go:67]:
```suggestion
	// If none is set, defaults to AWS waiter default which is 40 attempts.
```

Comment (sylviamoss) [reply builder/amazon/common/state.go:67]: is not me if there aren't any switched letters 😆

Comment (sylviamoss) [reply builder/amazon/common/state.go:67]: ops, I wanted to comment the other suggestion :D

Comment (azr) [reply builder/amazon/common/state.go]: yes, no worries and no worries :D

**Pull request merged**